### PR TITLE
[3D Image] Opt-in button in Image panel settings

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
None (feature flagged)

**Description**
When the new image panel is enabled, display a button in the **old** image panel to upgrade to the new panel. The button displays a confirmation dialog before doing the upgrade. Clicking "Replace panel" swaps in a new image panel.

The new panel does not have an option to toggle this back to "off". Instead the user has to click "Change panel" or add a new (legacy) Image panel.

image

image

Resolves FG-3213

### CODE REVIEW

File: packages/studio-base/src/panels/Image/ImageView.tsx [hunk 1]
```diff
@@ -283,6 +300,38 @@ export function ImageView({ context }: Props): JSX.Element {
     [onChangeCameraTopic],
   );
 
+  const upgradeConfirmDialog = useMemo(
+    () => (
+      <Dialog open={showUpgradeConfirmDialog} onClose={() => setShowUpgradeConfirmDialog(false)}>
+        <DialogTitle>Upgrade to the new Image panel?</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            This change will modify your layout. To revert to the old Image panel, you will need to
```
Comment (foxymiles) [inline packages/studio-base/src/panels/Image/ImageView.tsx:309]: Do we need to localize this?

File: packages/studio-base/src/panels/Image/ImageView.tsx [hunk 2]
```diff
@@ -283,6 +300,38 @@ export function ImageView({ context }: Props): JSX.Element {
     [onChangeCameraTopic],
   );
 
+  const upgradeConfirmDialog = useMemo(
```
Comment (foxymiles) [inline packages/studio-base/src/panels/Image/ImageView.tsx:303]: Can you just conditionally render this like `showUpgradeConfirmDialog && (<Dialog...`?

**Review by foxymiles — APPROVED** [review]
Code seems ok. When I tried this on the sample layout it didn't seem to switch cleanly. The image was blank and there was this error in settings:

Screenshot 2023-05-04 at 12 35 03 PM

**Comment by jtbandes** [comment]
> When I tried this on the sample layout it didn't seem to switch cleanly. The image was blank and there was this error in settings

This will be improved by #5882

**Comment by amacneil** [comment]
@defunctzombie to propose new dialog text
@2metres to propose new sidebar test

**Comment by amacneil** [comment]
New decision: skip this stage, and put `Image (Beta)` in the panel list for everyone when we are ready.

**Pull request closed without merging**